2600hz/kazoo-core
# KZOO-526 Delete attachment does not work if attachment is saved in aw…

## Solution text

Author: Egor Ezhov

Bug ID: KZOO-526 Delete attachment does not work if attachment is saved in aws s3

Details:

In the initial implementation (https://github.com/2600hz/kazoo-core/pull/372) the attachment is deleted based on the handler from the storage doc. It needs to be changed to use S3 data from the media doc which contains all required data.

The solution proposes:
- using S3 data from the media doc to delete attachment
- using attachment options to call `kzs_plan:plan` from the `kz_datamgr:delete_attachment` (same is used in the `put_attachment`, `fetch_attachment` and `update_attachment_metadata`) to correctly identify doc with media
- removing attachment's data from the `pvt_attachments` section in the media doc if the attachment is deleted. Data from the `_attachments` is deleted automatically by CouchDB. 
### Test Cases performed

1. Configured storage plan with AWS S3 for user data
2. Upload user's photo
3. Check that user's photo is stored in the AWS S3
4. Delete user's photo
5. Check that user's photo is deleted from the AWS S3.
6. Check that attachment's info is removed from the user doc
